fuel/src/main/kotlin/com/github/kittinunf/fuel/core/interceptors/RedirectionInterceptor.kt

```diff
@@ -1,34 +1,55 @@
 package com.github.kittinunf.fuel.core.interceptors
 
-import com.github.kittinunf.fuel.core.*
-import java.net.MalformedURLException
+import com.github.kittinunf.fuel.core.Encoding
+import com.github.kittinunf.fuel.core.FuelError
+import com.github.kittinunf.fuel.core.FuelManager
+import com.github.kittinunf.fuel.core.Method
+import com.github.kittinunf.fuel.core.Request
+import com.github.kittinunf.fuel.core.Response
+import java.net.URI
 import java.net.URL
 import javax.net.ssl.HttpsURLConnection
 
+const val HTTP_PERMANENT_REDIRECT = 308
+
 class RedirectException : Exception("Redirection fail, not found URL to redirect to")
 
 fun redirectResponseInterceptor(manager: FuelManager) =
         { next: (Request, Response) -> Response ->
             { request: Request, response: Response ->
-                if (response.statusCode == HttpsURLConnection.HTTP_MOVED_PERM ||
-                        response.statusCode == HttpsURLConnection.HTTP_MOVED_TEMP ||
-                        response.statusCode == 307   // 307 TEMPORARY REDIRECT - https://httpstatuses.com/307
-                        ) {
+
+                val newMethod = when (response.statusCode) {
+                    HttpsURLConnection.HTTP_MOVED_PERM,
+                    HttpsURLConnection.HTTP_MOVED_TEMP,
+                    HttpsURLConnection.HTTP_SEE_OTHER ->
+                        Method.GET
+                    else -> {
+                        request.method
+                    }
+                }
+
+                if (response.statusCode in HttpsURLConnection.HTTP_MULT_CHOICE..HTTP_PERMANENT_REDIRECT) {
```

Comment:
I think this range would be useful for reuse, would you consider defining it in HttpsUrlConnection?

Reply:
But we don't have HttpsUrlConnection object here though. When you said `defining it in HttpsUrlConnection?` it means extenstion of  `HttpsUrlConnection` ?

Reply:
yes unless you feel that is an error?

Reply:
sorry I see what you mean